### PR TITLE
Implement new thirdweb deploy flow

### PR DESCRIPTION
**Pull request by joaquim-verges**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   "scripts": {
     "clean": "rm -rf dist/",
     "dev": "tsdx watch",
-    "build:lib": "patch-package & rm -rf ./dist && tsup && yarn generate-types",
+    "build:lib": "rm -rf ./dist && tsup && yarn generate-types",
```
Comment (jnsdls) [inline package.json:23]: no longer need patching

File: package.json [hunk 2]
```diff
@@ -38,6 +38,7 @@
     "test:all": "ts-mocha --parallel --paths -t 120000 -r esm -p tsconfig.testing.json './test/**/*.test.ts'"
   },
   "devDependencies": {
+    "@esbuild-plugins/node-modules-polyfill": "^0.1.4",
```
Comment (jnsdls) [inline package.json:41]: using this for polyfilling for UMD because it works out of the box

File: package.json [hunk 3]
```diff
@@ -49,29 +50,24 @@
...
     "@types/expect": "^24.3.0",
     "@types/form-data": "^2.5.0",
-    "@types/lodash.isequal": "^4.5.5",
     "@types/mocha": "^9.1.1",
     "@types/node": "^17.0.0",
     "@types/uuid": "^8.3.3",
     "@typescript-eslint/eslint-plugin": "^5.2.0",
     "@typescript-eslint/parser": "^5.2.0",
     "chai": "^4.3.6",
+    "deep-equal": "^2.0.5",
```
Comment (jnsdls) [inline package.json:59]: I don't think we need this *in* the build, so dev deps should be fine (only used in tests)

File: package.json [hunk 4]
```diff
@@ -106,12 +99,5 @@
     "url": "https://github.com/thirdweb-dev/typescript-sdk/issues"
   },
   "homepage": "https://github.com/thirdweb-dev/typescript-sdk#readme",
-  "author": "thirdweb",
```
Comment (jnsdls) [inline package.json:107]: this was an extraneous leftover thing, all of this is defined in the tsup.config.ts file anyways

File: src/common/snapshots.ts [hunk 5]
```diff
@@ -32,12 +31,9 @@ export async function createSnapshot(
...
 
   const hashedLeafs = input.map((i) =>
-    hashLeafNode(
-      i.address,
-      ethers.utils.parseUnits(i.maxClaimable, tokenDecimals),
-    ),
+    hashLeafNode(i.address, utils.parseUnits(i.maxClaimable, tokenDecimals)),
   );
-  const tree = new MerkleTree(hashedLeafs, keccak256, {
+  const tree = new MerkleTree(hashedLeafs, utils.keccak256, {
```
Comment (jnsdls) [inline src/common/snapshots.ts:36]: ethers utils have keccak256, no need for the extraneous lib

File: src/contracts/vote.ts [hunk 6]
```diff
@@ -114,7 +113,7 @@ export class Vote implements UpdateableNetwork {
   public async get(proposalId: BigNumberish): Promise<Proposal> {
     const all = await this.getAll();
     const proposals = all.filter((p) =>
-      deepEqual(BigNumber.from(p.proposalId), BigNumber.from(proposalId)),
+      p.proposalId.eq(BigNumber.from(proposalId)),
```
Comment (jnsdls) [inline src/contracts/vote.ts:116]: @joaquim-verges let me know if there was a specific reason we were using deepEqual to compare 2 BigNumbers? they come with their own `eq` method so this was confusing to me while I was cleaning up dependencies (all tests still pass)

File: test/nft-drop.test.ts [hunk 7]
```diff
@@ -316,7 +313,7 @@ describe("NFT Drop Contract", async () => {
     const hashedLeafs = members.map((l) =>
       ethers.utils.solidityKeccak256(["address", "uint256"], [l, 0]),
     );
-    const tree = new MerkleTree(hashedLeafs, keccak256, {
+    const tree = new MerkleTree(hashedLeafs, ethers.utils.keccak256, {
```
Comment (jnsdls) [inline test/nft-drop.test.ts:316]: again, ethers has this for us, no need for a separate lib

File: test/token-drop.test.ts [hunk 8]
```diff
@@ -253,7 +250,7 @@ describe("Token Drop Contract", async () => {
     const hashedLeafs = members.map((l) =>
       ethers.utils.solidityKeccak256(["address", "uint256"], [l, 0]),
     );
-    const tree = new MerkleTree(hashedLeafs, keccak256, {
+    const tree = new MerkleTree(hashedLeafs, ethers.utils.keccak256, {
```
Comment (jnsdls) [inline test/token-drop.test.ts:253]: ethers utils again

File: tsup.config.ts [hunk 9]
```diff
@@ -1,7 +1,7 @@
-import alias from "esbuild-plugin-alias";
-import path from "path";
 import { defineConfig } from "tsup";
 
+import { NodeModulesPolyfillPlugin } from "@esbuild-plugins/node-modules-polyfill";
```
Comment (jnsdls) [inline tsup.config.ts:3]: this replaces the inject, the alias, some of the global banner, very nice

File: tsup.config.ts [hunk 10]
```diff
@@ -17,33 +17,22 @@ export default defineConfig([
...
+    platform: "browser",
+    replaceNodeEnv: true,
+    globalName: "_thirdweb",
+    format: ["iife"],
+    banner: {
+      js: "window.global=window;window.globalThis=window;",
+    },
+    // inject ThirdwebSDK into window
+    footer: { js: "window.ThirdwebSDK = window._thirdweb.ThirdwebSDK;" },
+    esbuildPlugins: [NodeModulesPolyfillPlugin()],
```
Comment (jnsdls) [inline tsup.config.ts:36]: so much cleaner

**Review by jnsdls — COMMENTED** [review]
@joaquim-verges check this out in the morning :)

**Comment by jnsdls** [comment]
tests don't pass because some abi json files required in tests are not checked in :)

File: package.json (hunk 4 above)
Comment (joaquim-verges) [reply package.json:107]: isnt that what gets picked up by npm though? like to show on our npm page?

File: src/contracts/vote.ts (hunk 6 above)
Comment (joaquim-verges) [reply src/contracts/vote.ts:116]: looks like an old leftover - def not needed here

**Review by jnsdls — APPROVED** [review]
LGTM

**Pull request merged**